plugins/renderer-inline-row-separators/test/inline_row_separators_test.mocha.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+require('../src/inline_row_separators_renderer');
+const chai = require('chai');
+const Blockly = require('blockly');
+const jsdomGlobal = require('jsdom-global');
+const {addInlineRowSeparators, overrideOldBlockDefinitions} =
+    require('../src/index');
+
+overrideOldBlockDefinitions();
+
+const assert = chai.assert;
+
+suite('inlineRowSeparates', function() {
+  setup(function() {
+    this.jsdomCleanup =
+        jsdomGlobal('<!DOCTYPE html><div id="blocklyDiv"></div>');
+    this.workspace = Blockly.inject('blocklyDiv', {
+      renderer: 'thrasos-inline-row-separators',
+    });
+    this.renderer = this.workspace.getRenderer();
+    Blockly.defineBlocksWithJsonArray([
+      {
+        'type': 'test_block',
+        'message0': 'input A %1 %2 input B %3',
+        'args0': [
+          {'type': 'input_value', 'name': 'A'},
+          {'type': 'input_dummy'},
+          {'type': 'input_value', 'name': 'B'},
+        ],
+        'inputsInline': true,
+      },
+    ]);
+  });
+
+  teardown(function() {
+    delete Blockly.Blocks['test_block'];
+    this.workspace.dispose();
+    this.jsdomCleanup();
+  });
+
+  test('shouldStartNewRow_ returns true if last input is a dummy', function() {
+    const block = this.workspace.newBlock('test_block');
+    const renderInfo = this.renderer.makeRenderInfo_(block);
+    // Test that a dummy after a value still remains on one row.
+    assert.isFalse(
+        renderInfo.shouldStartNewRow_(block.inputList[1], block.inputList[0]));
+    // Test that a value after a dummy is put on a new row.
+    assert.isTrue(
+        renderInfo.shouldStartNewRow_(block.inputList[2], block.inputList[1]));
+  });
+
+  test('inline_text_join is inline and inserts dummies', function() {
+    const block = this.workspace.newBlock('inline_text_join');
+    assert.isTrue(block.inputList[0].type === Blockly.inputTypes.DUMMY);
+    assert.isTrue(block.inputList[1].type === Blockly.inputTypes.VALUE);
+    assert.isTrue(block.inputList[2].type === Blockly.inputTypes.DUMMY);
+    assert.isTrue(block.inputList[3].type === Blockly.inputTypes.VALUE);
+    const renderInfo = this.renderer.makeRenderInfo_(block);
+    assert.isTrue(renderInfo.isInline);
```

Comment:
What is this assertion guarunteeing? Is it to ensure that the base `shouldStartNewRow_` maintains its current behavior? Could you leave a comment on one of these assertions explaining why you're asserting it? (no need to duplicate the comment to all of them).

Reply:
This is just testing that the structure of the new demonstration block definitions meet the requirements to demonstrate the new feature, in other words that they insert dummies between values and enable inline input mode, unlike the old blocks. The behavior of the renderer is tested elsewhere, and I've stopped referring to it here now in favor of checking the block's inputsInline property directly. I've added some comments.